### PR TITLE
Existing types cleanup + more api methods onto new request method

### PR DESCRIPTION
**Pull request by FredKSchott**
- This PR ended up tackling **a lot** of type clean up and organization
- getPaymentChannel() now uses this.request()
- getSettings() now uses this.request()
- getLedger() now uses this.request()
- existing transaction types (`ledger/types.ts`, `ledger/transaction-types.ts`, etc.) cleaned up, but not yet completely moved to new system

/cc @intelliot

### CODE REVIEW

File: src/common/types/objects/ledger_entries.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import {SignerEntry} from './index'
+
+export type PayChannelLedgerEntry = {
```
Comment (intelliot) [inline src/common/types/objects/ledger_entries.ts]: Why use `type` here instead of `interface`?

**Comment by sublimator** [comment]
Might be a leftover from Flow syntax :)

**Comment by intelliot** [comment]
Superseded by #860

File: src/common/types/objects/ledger_entries.ts (hunk 1 above)
Comment (FredKSchott) [reply src/common/types/objects/ledger_entries.ts]: @sublimator is right, that's a left over from Flow. Changing to interface...

**Comment by FredKSchott** [comment]
@sublimator is right, that's a left over from Flow. Changing to interface...

**Pull request merged**